sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -200,31 +203,21 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
    * @param locationId location id
    * @returns InstitutionLocation
    */
-  async getLocationById(locationId: number): Promise<InstitutionLocation> {
-    return this.repo
-      .createQueryBuilder("location")
-      .select(["location.name", "location.id"])
-      .where("location.id = :locationId", { locationId })
-      .getOne();
-  }
-
-  /**
-   * Get institution location by location id.
-   * @param locationId location id
-   * @returns InstitutionLocation
-   */
-  async getDesignatedLocationById(
+  async getLocation(
     locationId: number,
-  ): Promise<InstitutionLocation> {
+  ): Promise<DesignatedAndNotDesignatedLocations> {
     return this.repo
       .createQueryBuilder("location")
-      .select(["location.name"])
-      .where("location.id = :locationId", { locationId })
-      .andWhere(
-        `EXISTS(${this.designationAgreementLocationService
-          .getExistsDesignatedLocation()
-          .getSql()})`,
+      .select("location.name", "locationName")
+      .addSelect(
```

Comment:
Because we are using the addSelect I would prefer to use the overload, as below:
```typescript
.addSelect(
        `CASE
          WHEN EXISTS(${this.designationAgreementLocationService
            .getExistsDesignatedLocation()
            .getSql()}) THEN true
          ELSE false
        END`, "isDesignated"
      )
```

Reply:
ya... 👍